### ISSUE TITLE
random failure on startup with assert fail

### ISSUE THREAD

**Issue by dzmitry-lahoda** [smol-dot/smoldot#490]
```
[2023-04-30T16:53:36Z WARN  sync-service-polkadot] GrandPa warp sync in progress. Block: #15319560 (0x0xcc6c…806c). Peer attempt: 12D3KooWSRjL9LcEQd5u2fQTbyLxTEHq1tUFgQ6amXSp8Eu7TfKP.
[2023-04-30T16:53:37Z INFO  sync-service-polkadot] GrandPa warp sync finished to #15319560 (0xcc6c…806c)
[2023-04-30T16:53:37Z INFO  runtime-polkadot] Finalized block runtime ready. Spec version: 9370. Size of `:code`: 1.3 MiB.
thread 'async-std/runtime' panicked at 'assertion failed: `(left == right)`
  left: `0`,
 right: `1`', /home/dz/.cargo/git/checkouts/smoldot-2dcbf637e11a77d4/e4ebb26/light-base/src/sync_service/parachain.rs:443:41
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'async-std/runtime' panicked at 'called `Result::unwrap()` on an `Err` value: Canceled', /home/dz/.cargo/git/checkouts/smoldot-2dcbf637e11a77d4/e4ebb26/light-base/src/sync_service.rs:213:18
```

**Comment by tomaka** [comment]
These are two different panics, so I've opened another issue.

**Comment by tomaka** [comment]
> I am ok if instead of fixes at least better unwraps with expect.

`expect` doesn't actually bring anything compared to `unwrap`. The panic message already contains the line number, which is more than enough to figure what the assumption was. Substrate has a rule to use `expect`, but that's a way to force programmers to not just put `unwrap` without thinking, not because `expect` is inherently better.

**Comment by tomaka** [comment]
> I can turn on logs, but as issues are hard to repro, can do it overnight or something and share big file in archieval. If that helps.

This issue (in parachain.rs) was easy to fix, but the other one (#492) looks more tricky. I can't tell right now if logs would be helpful, but if it's easy for you to generate them, it can't hurt :pray:
